flink-java/src/main/java/org/apache/flink/api/java/functions/SelectByMinFunction.java

```diff
@@ -72,8 +73,8 @@ public T reduce(T value1, T value2) throws Exception {
 		for (int position : fields) {
 			// Save position of compared key
 			// Get both values - both implement comparable
-			Comparable comparable1 = value1.getFieldNotNull(position);
-			Comparable comparable2 = value2.getFieldNotNull(position);
+			Comparable comparable1 = ((Tuple)value1).getFieldNotNull(position);
```

Comment:
Yes, since it was intended to work with Flink's Java tuples. But a Scala tuple is not a subtype of `Tuple`. Thus, I would assume that you can't reuse the Java implementation here.